lib/daemon/worker.js

```diff
@@ -20,24 +20,21 @@ const wrapMethod = (action, compressor) => args =>
         if (err) {
           return reject(err);
         }
-        compressor.saveCache();
         return resolve(result);
       }
     ));
 
 // Create each action method, injecting a compressor
 // instance for each one.
-Promise.all(
-  Object.keys(actions.ACTIONS).map(action =>
-    Compressor.init({
-      scope: `${action}-compressor`,
-      cacheDir
-    }).then(compressor => ({
-      [action]: wrapMethod(action, compressor)
-    })))
-)
-  .then(actionMaps =>
-    actionMaps.reduce((acc, map) =>
-      Object.assign({}, acc, map), {})
+workerpool.worker(
+  Object.keys(actions.ACTIONS)
+    .map(action => ({
```

Comment:
Is the `.map` needed? You could just reduce.

Reply:
But if it reads clearer to you broken apart, that's fine.

Reply:
Nah, makes plenty of sense as a single reduce!